clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2992,31 +3001,40 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
     return Call.getAs<Expr>();
   }
 
-  // Adds default initializer for generated type and creates
-  // a call to __builtin_memcpy to initialize local clone from
-  // kernel argument.
   void handleGeneratedType(FieldDecl *FD, QualType Ty) {
-    addFieldInit(FD, Ty, None,
-                 InitializationKind::CreateDefault(KernelCallerSrcLoc));
-    addFieldMemberExpr(FD, Ty);
-    Expr *ParamRef = createGetAddressOf(createParamReferenceExpr());
-    Expr *LocalCloneRef = createGetAddressOf(MemberExprBases.back());
-    Expr *MemCpyCallExpr = buildMemCpyCall(ParamRef, LocalCloneRef, Ty);
```

Comment:
Yes, I think I can do that now. I'm also thinking of moving generation of ReinterpretCastExpr to a function, since there is a slight duplication of code in `handleGeneratedType` for fields and bases.

Reply:
Okay, done.